diff -uNr scsiinfo-1.7.orig/Makefile scsiinfo-1.7/Makefile
--- scsiinfo-1.7.orig/Makefile	Sat Aug 23 15:12:46 1997
+++ scsiinfo-1.7/Makefile	Tue Feb 29 22:34:50 2000
@@ -1,28 +1,30 @@
-CFLAGS=-O2 -fomit-frame-pointer -Wall -fno-strength-reduce
+CFLAGS=-Wall -D_GNU_SOURCE $(OPT)
 #CFLAGS=-g -Wall -fno-strength-reduce
 LDFLAGS=-s
 
 #do not modify this without changing the tcl/tk scripts
 
 BINDIR=/usr/lib/scsi
-MANDIR=/usr/man
+MANDIR=/usr/share/man
 WISHEXECS=tk/cache tk/control tk/disconnect tk/error tk/format\
 	tk/inquiry tk/notch tk/peripheral tk/rigid tk/save-changes tk/verify\
 	tk/save-file tk/overview
 TARGETS=scsiinfo scsiformat tworands
+DOC = 0-CHANGES 0-README.first 0-TODO
 
 .PHONY: all install dist clean
 
 all: $(TARGETS)
 
 install: $(TARGETS) $(WISHEXECS) scsi-config tk_scsiformat
-	cp scsiinfo scsiformat /usr/bin
-	if [ ! -d $(BINDIR) ]; then mkdir -p $(BINDIR); fi
-	cp tworands $(BINDIR)
-	install-wish $(BINDIR) $(WISHEXECS)
-	install-wish /usr/bin scsi-config tk_scsiformat
-	cp tk/generic $(BINDIR)
-	cp man8/* $(MANDIR)/man8
+	install -s -m 755 scsiinfo scsiformat $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)$(BINDIR)
+	install -s -m 755 tworands $(DESTDIR)$(BINDIR)
+	./install-wish $(DESTDIR)$(BINDIR) $(WISHEXECS)
+	./install-wish $(DESTDIR)/usr/bin scsi-config tk_scsiformat
+	install -m 644 tk/generic $(DESTDIR)$(BINDIR)
+	install -m 644 man8/* $(DESTDIR)$(MANDIR)/man8
+	install -m 755 sgcheck $(DESTDIR)/usr/sbin
 
 clean:
 	rm -f core *~ *.o $(TARGETS)
