--- ./Makefile.org	Sun Jan 13 00:10:07 2002
+++ ./Makefile	Sun Jan 13 00:17:19 2002
@@ -2,7 +2,7 @@
 
 PREFIX=/usr/local
 INSTDIR=$(DESTDIR)/$(PREFIX)/bin
-MANDIR=$(DESTDIR)/$(PREFIX)/man
+MANDIR=$(DESTDIR)/$(PREFIX)/share/man
 
 CC = gcc
 LD = gcc
@@ -17,7 +17,7 @@
 MAN_PGS = sg_dd.8 sgp_dd.8 sg_map.8 sg_rbuf.8
 MAN_PREF = man8
 
-CFLAGS = -g -O2 -Wall -D_REENTRANT
+CFLAGS = $(OPT) -Wall -D_REENTRANT
 # CFLAGS = -g -O2 -Wall -D_REENTRANT -DSG_KERNEL_INCLUDES
 # CFLAGS = -g -O2 -Wall -pedantic -D_REENTRANT
 
@@ -91,10 +91,10 @@
 install: $(EXECS) $(COMMON)
 	install -d $(INSTDIR)
 	for name in $^; \
-	 do install -s -o root -g root -m 755 $$name $(INSTDIR); \
+	 do install -s -m 755 $$name $(INSTDIR); \
 	done
 	for mp in $(MAN_PGS); \
-	 do install -o root -g root -m 644 $$mp $(MANDIR)/$(MAN_PREF); \
+	 do install -s -m 644 $$mp $(MANDIR)/$(MAN_PREF); \
 	 gzip -9f $(MANDIR)/$(MAN_PREF)/$$mp; \
 	done
 
